### PR TITLE
feature refactoring the code

### LINKED ISSUE #7: [Feature]: Refactoring the code
### Description

Simon's code is in a jupyter notebook. 
It must be put into .py files. 
The jupyter notebooks can be used for experimentation but the main code should not be put into them.

### Expected behaviour

1 file for each main component, containing (at least) one class. 
- [ ] Agent file: 
   - [ ] action selection
   - [ ] recording of state, action, reward, new_state, done (a dedicated Memory object could be a good idea)
   - [ ] training of the Q network 
      - [ ] collect of a batch of previous experiences
      - [ ] generation of the training signal
      - [ ] run a train of the Q net
    
- [ ] Q Network file
  - [ ] pytorch NN class

- [ ] Trainer file
  - [ ] training
  - [ ] testing
  - [ ] saving the params / hyper params 

- [ ] Preprocessing file (input data consists of arrays of string, it should be preprocessed before ingestion by the Q Net)
  - [ ] Convert the strings into numerical data
     - [ ] ordinal encoding ?
     - [ ] one hot encoding
  - [ ] Convert the encoded data into torch compatible data 

### Environment

- OS and version

### Proposed solution

_No response_

### LINKED ISSUE #8: [Bug]: Agent performances are not good
### Description

@STTR13 told us that the agent performs not very well after a long training. It should be fixed.

### What happened?

A bug happened!

### Expected behaviour

A working agent.

### Environment

- OS and version
- Branch that causes this bug

### Steps to reproduce

- Train the agent.
- Test it.

### Relevant log output

_No response_

### Proposed solution

_No response_

### PR DESCRIPTION
**Pull request by staeter**
# Description

We've refactored the code, fixed some small algorithmic issues and added the students names and pictures to the readme.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the needed labels
- [x] I have linked this PR to an issue
- [x] I have linked this PR to a milestone
- [x] I have linked this PR to a project
- [x] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [x] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Review by ezalos — APPROVED** [review]

**Review by qfeuilla — APPROVED** [review]
LGTM
